http/events.go

```diff
@@ -55,6 +55,7 @@ func (sub *eventSubscriber) handleEventsSubscribeWebsocket() (websocket.StatusCo
 		return websocket.StatusUnsupportedData, "Error subscribing", nil
 	}
 	defer cancel()
+	logger.Debug("WebSocket is subscribed to messages", "namespaces", sub.namespacePatterns, "event_types", sub.pattern, "bexpr_filter", sub.bexprFilter)
```

Comment:
By the time we to this code (where we synchronously create the subscription on the event bus), we've already called `websocket.Accept` here:

https://github.com/hashicorp/vault/blob/5e5644714e219b983ecfb2ba377073603b52f673/http/events.go#L225-L230

I wonder if we could eliminate the race condition for all clients and not just tests by reversing that order?

Reply:
Oh, that's a good idea. Let me investigate that.

Reply:
Done. I had to rework the code to avoid a data race detector false positive, but I think this is a lot better. Thanks for the suggestion!